.github/workflows/client.yaml

```diff
@@ -0,0 +1,27 @@
+name: Go Client Tests
+
+on:
+  push:
+    branches: ["*"]
+  pull_request:
+    branches: [$default-branch]
```

Comment:
I can't find this in the docs, have you used this before?

Reply:
see https://github.com/actions/starter-workflows/blob/main/ci/go.yml

Reply:
No wait the actions are reacting on the push action not pull_request. Change branches to the `$default-branch`

Reply:
I wanted to have tests run before opening a PR. Obviously that makes tests on PR redundant, but I was not sure if they show up inside the PR as well.